modules/library/main/src/main/java/org/geotools/data/ows/URLChecker.java

```diff
@@ -0,0 +1,31 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2020, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.data.ows;
+
+/**
+ * Defines how to implement URL validation
+ *
+ * @author ImranR
+ */
+public interface URLChecker {

```

Comment:
Please add javadocs for all public methods, this is kind of the most important interface in this work, and has no docs.

Reply:
will do